tasks/broad/ImputationTasks.wdl

```diff
@@ -0,0 +1,752 @@
+  }
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+}
+
+task SelectVariantsByIds {
+  input {
+    File vcf
+    File ids
+    String basename
+    String gatk_docker
+  }
+
+  Int disk_size = ceil(1.2*size(vcf, "GiB")) + 100
+
+  parameter_meta {
+    vcf: {
+         description: "vcf",
+         localization_optional: true
+       }
+  }
+
+  command <<<
+    cp ~{ids} sites.list
+    gatk SelectVariants -V ~{vcf} --exclude-filtered --keep-ids sites.list -O ~{basename}.vcf.gz
+  >>>
+
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " SSD"
+    memory: "16 GiB"
+  }
+
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task RemoveAnnotations {
+  input {
+    File vcf
+    String basename
+    String bcftools_docker
+  }
+
+  Int disk_size = ceil(2.2*size(vcf, "GiB")) + 100
+
+  command <<<
+    bcftools annotate ~{vcf} -x FORMAT,INFO -Oz -o ~{basename}.vcf.gz
+    bcftools index -t ~{basename}.vcf.gz
+  >>>
+
+  runtime {
+    docker: bcftools_docker
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task InterleaveVariants {
+  input {
+    Array[File] vcfs
+    String basename
+    String gatk_docker
+  }
+
+  Int disk_size = ceil(3.2*size(vcfs, "GiB")) + 100
+
+  command <<<
+    gatk MergeVcfs -I ~{sep=" -I " vcfs} -O ~{basename}.vcf.gz
+  >>>
+
+
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk " + disk_size + " SSD"
+    memory: "16 GiB"
+  }
+
+  output {
+    File output_vcf = "~{basename}.vcf.gz"
+    File output_vcf_index = "~{basename}.vcf.gz.tbi"
+  }
+}
+
+task FindSitesUniqueToFileTwoOnly {
+  input {
+    File file1
+    File file2
+    String ubuntu_docker
+  }
+
+  Int disk_size = ceil(size(file1, "GiB") + 2*size(file2, "GiB")) + 100
+
+  command <<<
+    comm -13 <(sort ~{file1} | uniq) <(sort ~{file2} | uniq) > missing_sites.ids
+  >>>
+
+  runtime {
+    docker: ubuntu_docker
+    disks: "local-disk " + disk_size + " HDD"
+    memory: "4 GiB"
+  }
+
+  output {
+    File missing_sites = "missing_sites.ids"
+  }
+}
+
+task SplitMultiSampleVcf {
+  input {
+    File multiSampleVcf
+    Int mem = 8
+    Int bcftools_docker
+  }
+
+  Int disk_size = ceil(3*size(multiSampleVcf, "GiB")) + 100
+
+  command <<<
+    mkdir out_dir
+    bcftools +split ~{multiSampleVcf} -Oz -o out_dir
+    for vcf in out_dir/*.vcf.gz; do
+      bcftools index -t $vcf
+    done
+  >>>
+
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk " + disk_size + " SSD"
+    memory: mem + " GiB"
+  }
+
+  output {
+    Array[File] single_sample_vcfs = glob("out_dir/*.vcf.gz")
+    Array[File] single_sample_vcf_indices = glob("out_dir/*.vcf.gz.tbi")
+  }
+}
+
+task CrosscheckFingerprints {
```

Comment:
This may overlap with other CrosscheckFingerpirnt task(s) we have in Warp. We should consider consolidating

Reply:
👍